extensions/amp-access/0.1/amp-access.js

```diff
@@ -594,9 +601,10 @@ export class AccessService {
     }
 
     log.fine(TAG, 'Start login');
-    const urlPromise = this.buildUrl_(assert(this.config_.login,
-        'Login URL is not configured'), /* useAuthData */ true);
-    this.loginPromise_ = this.openLoginDialog_(urlPromise).then(result => {
+    assert(this.config_.login, 'Login URL is not configured');
+    // Login URL should always be available at this time.
+    const loginUrl = assert(this.loginUrl_);
```

Comment:
I don't think there's any guarantee `loginUrl_` has been set yet... How about chaining off a `loginUrlPromise` (that'll ensure that the login url has resolved)?

```js
this.loginUrlPromise_.then((url) => {
  return this.openLoginDialog_(url);
}).then((result) => {
  // ...
})